### PR TITLE
Grading table improvements (filter by name, username, group)

### PR DESCRIPTION
**Pull request by RichDom2185**
### Description

No BE change required.

Adds resizable columns, filter by name, username, group.

image

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Code quality improvements

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

File: src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx [hunk 1]
```diff
@@ -410,6 +414,25 @@ const GradingSubmissionTable: React.FC<GradingSubmissionTableProps> = ({
...
+            }
+            const filters = e.api.getFilterModel();
+            const cellFilters = [];
+            for (const [key, { filter: query }] of Object.entries(filters)) {
+              switch (key) {
+                // Fields that BE supports filtering on
+                case ColumnFields.studentName:
+                case ColumnFields.studentUsername:
+                case ColumnFields.groupName:
+                  console.log(query);
```
Comment (sayomaki) [inline src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx:429]: Do we need this here?

**Review by sayomaki — COMMENTED** [review]
Minor nit on the logging part, but otherwise the code looks great!

File: src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx (hunk 1 above)
Comment (RichDom2185) [reply src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx:429]: Oops

Comment (RichDom2185) [reply src/pages/academy/grading/subcomponents/GradingSubmissionsTable.tsx]: Resolved in 59421b029fe484b126e584196b3e4b39322a0e7a.

**Review by sayomaki — APPROVED** [review]
Looks good to be merged!

**Pull request merged**